adagrams/game.py

```diff
@@ -1,13 +1,137 @@
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+SCORE_CHART = {
+1 : ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T' ],
+2 : ['D', 'G'],
+3 : ['B', 'C', 'M', 'P'],
+4 : ['F', 'H', 'V', 'W', 'Y'],
+5 : ['K'],
+8 : ['J', 'X'],
+10 : ['Q', 'Z']
+}
 def draw_letters():
-    pass
+    """
+    This function takes no argument and return a list of 10 strings
+    output: list of 10 strings
+    """
+  
+    letter_list = []
+    player_hand = []
+    freq_list = []
+
+    # loop to store the key and value of LETTER_POOL to separate list
+    for key in LETTER_POOL:
+        letter_list.append(key)
+        freq_list.append(LETTER_POOL[key])
+
+    #This loop helps to generate a hand of 10 strings and add it to a list
+    for item in range(10):
+        random_index_of_letter_list = randint(0,len(letter_list)-1)
+        letter_picked = letter_list[random_index_of_letter_list]
+        player_hand.append(letter_picked)
+        freq_list[random_index_of_letter_list] -= 1
+        if freq_list[random_index_of_letter_list] == 0:
+            letter_list.pop(random_index_of_letter_list)
+            freq_list.pop(random_index_of_letter_list)
+      
+    return player_hand            
+    
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    This function checks whether the word passed has characters that are
+    contained within a collection (or hand) of drawn letters
+    input : variable word as a string, list of drawn letters
+    output: True if all characters of the word are present in 
+            the list or else false
+    """
+    
+    word = word.upper() # To convert all the characters to uppercase.
+    letter_bank_copy = letter_bank.copy()
+    for character in word:
+        if character in letter_bank_copy:
+            letter_bank_copy.remove(character)
+        else:
+            return False
```

Comment:
I'd prefer to flip this logic around so that you check if a letter is not in `letter_bank_copy` and then immediately return `False`. 

```suggestion
        if character not in letter_bank_copy:
            return False
        letter_bank_copy.remove(character)
```

This allows you to remove `else`, which matters because very line of code is another place where a bug can be created, it reduces the conditional complexity of your code, and can highlight to other devs that the the main logic of the loop is to find a letter and remove it.